### PR TITLE
LPS-158005 [Raylife AP] Drive Info Validations

### PR DESCRIPTION
**Pull request by eliasantosf**
Sub-Task: [LPS-158005](https://issues.liferay.com/browse/LPS-158005)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 1]
```diff
@@ -47,42 +57,147 @@ const DriverInfoForm = ({
...
 	);
 
 	const accidentCitations = form.map((form) => {
-		return form.accidentCitation.filter(
-			(accidentCitation: {id: number}) => accidentCitation.id !== id
+		const list = form.accidentCitation || [];
+
+		return list.filter(
+			(accidentCitation: {id: number}) =>
+				accidentCitation.id !== form.accidentCitation.id
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:135]: What does "list" mean? Try something like: 
```suggestion
		const listOfAccidentCitations = form.accidentCitation || [];

		return listOfAccidentCitations.filter(
			(accidentCitation: {id: number}) =>
				accidentCitation.id !== form.accidentCitation.id
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 2]
```diff
@@ -47,42 +57,147 @@ const DriverInfoForm = ({
 
 	const [occupationValue, setOccupationValue] = useState<string>();
 
+	const [options, setOptions] = useState<any[]>([]);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:60]:
```suggestion
	const [newAccidentCitation, setNewAccidentCitation] = useState<any[]>([]);
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! I just added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 3]
```diff
@@ -47,42 +57,147 @@ const DriverInfoForm = ({
...
+		maritalStatus: false,
+		millitaryAffiliation: false,
+		occupation: false,
+		otherOccupation: false,
+		relationToContact: false,
+	};
+
+	const [hasError, setHasError] = useState(hasRequiredError);
+
+	const ErrorMessage = ({text}: any) => (
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:114]: Isn't this supposed to be a string? Like this: 
```suggestion
	const ErrorMessage = ({text: string}) => (
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 4]
```diff
@@ -369,68 +289,134 @@ const DriverInfoForm = ({
...
+											: false,
+								});
 							}}
 							required
 							type="text"
+							value={form[formNumber - 1].lastName}
 						/>
 
-						<label htmlFor="lastName">
+						<label htmlFor="lasstName">
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:380]: Please takeout the extra 's'
```suggestion
						<label htmlFor="lastName">
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 5]
```diff
@@ -553,59 +599,103 @@ const DriverInfoForm = ({
 						</ClaySelect>
 
 						<label htmlFor="occupation">
-							Occupation&nbsp;
+							Ocuppation&nbsp;
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:602]: Please takeout the extra 'p' and add back 'c'
```suggestion
							Occupation&nbsp;
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 6]
```diff
@@ -553,59 +599,103 @@ const DriverInfoForm = ({
...
 							required
 							type="text"
+							value={form[formNumber - 1].otherOccupation}
 						/>
 
-						{occupationValue === 'other' && (
+						{occupationValue === 'Other' && (
 							<label htmlFor="otherOccupation">
-								Other Occupation
+								Other Ocupation
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:653]: Please add back the 'c'
```suggestion
								Other Occupation
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 7]
```diff
@@ -553,59 +599,103 @@ const DriverInfoForm = ({
...
 								<span className="text-danger-darken-1">*</span>
 							</label>
 						)}
 
-						{occupationValue !== 'other' && (
+						{occupationValue !== 'Other' && (
 							<label htmlFor="otherOccupation">
-								Other Occupation
+								Other Ocupation
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:660]: Please add back the 'c'
```suggestion
								Other Occupation
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/index.tsx [hunk 8]
```diff
@@ -63,6 +58,97 @@ const DriverInfo = () => {
...
+
+		dispatch({
+			payload: true,
+			type: ACTIONS.SET_IS_ABLE_TO_SAVE,
+		});
+		dispatch({
+			payload: hasAllRequiredFieldsToNextStep,
+			type: ACTIONS.SET_IS_ABLE_TO_NEXT,
+		});
+	};
```
Comment (JoMaAlves) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/index.tsx:136]: I believe that we can substitute the `.every` comparison for a single function that does that, maybe we could even turn these 3 functions into one. I would like to hear @silasbsi 's opinion on this.

**Review by JoMaAlves — COMMENTED** [review]
LGTM 🔥 But added some corrections.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx (hunk 3 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:114]: This attribute must be any.

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by joanagrego — APPROVED** [review]
Great job! :arrow_up:

**Comment by silasbsi** [comment]
ci:forward

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**